app/atomic_svc.py

```diff
@@ -0,0 +1,234 @@
+
+        with open(enterprise_attack_path, 'r') as f:
+            mitre_json = json.load(f)
+
+        for phase_name, external_id in self._gen_single_match_tactic_technique(mitre_json):
+            self.technique_to_tactics[external_id].append(phase_name)
+
+    def _handle_attachment(self, attachment_path):
+        # attachment_path must be a POSIX path
+        payload_name = os.path.basename(attachment_path)
+        # to avoid collisions between payloads with the same name
+        with open(attachment_path, 'rb') as f:
+            h = hashlib.md5(f.read()).hexdigest()
+        payload_name = h[:PREFIX_HASH_LEN] + '_' + payload_name
+        shutil.copyfile(attachment_path, os.path.join(self.payloads_dir, payload_name), follow_symlinks=False)
+        return payload_name
+
+    @staticmethod
+    def _normalize_path(path, platform):
+        if platform == PLATFORMS['windows']:
+            return path.replace('\\', '/')
+        return path
+
+    def _catch_path_to_atomics_folder(self, string_to_analyse, platform):
+        """
+        Catch a path to the atomics/ folder in the `string_to_analyse` variable,
+        and handle it in the best way possible. If needed, will import a payload.
+        """
+        regex = re.compile(r'\$?PathToAtomicsFolder((?:/[^/ \n]+)+|(?:\\[^\\ \n]+)+)')
+        payloads = []
+        if regex.search(string_to_analyse):
+            fullpath, path = regex.search(string_to_analyse).group(0, 1)
+            path = self._normalize_path(path, platform)
+
+            # take path from index 1, as it starts with /
+            path = os.path.join(self.repo_dir, 'atomics', path[1:])
+
+            if os.path.isfile(path):
+                payload_name = self._handle_attachment(path)
+                payloads.append(payload_name)
+                string_to_analyse = string_to_analyse.replace(fullpath, payload_name)
+
+        return string_to_analyse, payloads
+
+    def _use_default_inputs(self, test, platform, string_to_analyse):
+        """
+        Look if variables are used in `string_to_analyse`, and if any variable was given
+        a default value, use it.
+        """
+        payloads = []
+        defaults = dict((key, val) for key, val in test.get('input_arguments', dict()).items())
+        while RE_VARIABLE.search(string_to_analyse):
+            full_var_str, varname = RE_VARIABLE.search(string_to_analyse).groups()
+            default_var = str(defaults.get(varname, dict()).get('default'))
+
+            if default_var:
+                default_var, new_payloads = self._catch_path_to_atomics_folder(default_var, platform)
+                payloads.extend(new_payloads)
+                string_to_analyse = string_to_analyse.replace(full_var_str, default_var)
+
+        return string_to_analyse, payloads
+
+    @staticmethod
+    def _handle_multiline_commands(cmd):
+        return cmd.replace('\n', ';')
+
+    async def _prepare_cmd(self, entries, test, platform, cmd):
```

Comment:
entries variable appears unused, let's not pass it in

Reply:
fixed in 0948d05